quadriga/src/main/java/edu/asu/diging/quadriga/web/DisplayCollectionController.java

```diff
@@ -0,0 +1,95 @@
+package edu.asu.diging.quadriga.web;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.bson.types.ObjectId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import edu.asu.diging.quadriga.core.exceptions.CollectionNotFoundException;
+import edu.asu.diging.quadriga.core.exceptions.InvalidObjectIdException;
+import edu.asu.diging.quadriga.core.model.Collection;
+import edu.asu.diging.quadriga.core.model.EventGraph;
+import edu.asu.diging.quadriga.core.model.MappedTripleGroup;
+import edu.asu.diging.quadriga.core.service.CollectionManager;
+import edu.asu.diging.quadriga.core.service.EventGraphService;
+import edu.asu.diging.quadriga.core.service.MappedTripleGroupService;
+
+@Controller
+public class DisplayCollectionController {
+
+    @Autowired
+    private CollectionManager collectionManager;
+
+    @Autowired
+    private EventGraphService eventGraphService;
+    
+    @Autowired
+    private MappedTripleGroupService mappedTripleGroupService;
+    
+    private Logger logger = LoggerFactory.getLogger(getClass());
+
+    @RequestMapping(value = "/auth/collections/{id}", method = RequestMethod.GET)
+    public String get(@PathVariable String id, Model model) {
+        
+        Collection collection;
+        try {
+            collection = collectionManager.findCollection(id);
+            if(collection == null) {
+                return "error404Page";
+            }
+        } catch (InvalidObjectIdException e) {
+            return "error404Page";
+        }
+        
+        ObjectId collectionId = collection.getId();
+        
+        List<EventGraph> eventGraphs = eventGraphService.findAllEventGraphsByCollectionId(collectionId);
```

Comment:
since here only the latest event graph is required, I would add a new manager method that gets only. the latest

Reply:
But the list of EventGraphs is also being used later in the Controller to get the number of submitted networks where one network can have multiple EventGraphs
Should I add two methods in the manager, one that gives latest event Graph and one that gives number of submitted networks?
I think this list is also being used for the next story (Q02-19)

Reply:
I think the assumption is wrong that there are multiple eventGraphs for a network. There should be only one networks=eventGraph. and if I'm right, then yes, there should be one method to get the latest and one to count.

Reply:
Actually as per the existing code, one network will have multiple EventGraphs as per https://github.com/diging/quadriga-2.0/blob/develop/quadriga/src/main/java/edu/asu/diging/quadriga/api/v1/AddNetworkApiController.java#L54

What happens here is, for every "start" node in the network, an eventGraph is created
I also tested this logic with custom input and 3 EventGraphs were created by this logic for 1 network
I also verified this logic in old quadriga code, where, for displaying a network, multiple EventGraphs are merged by finding common nodes between the EventGraphs: https://github.com/diging/quadriga/blob/develop/Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/EventParser.java

Reply:
Alright that makes sense, we don't care about the number of networks though. Just about how many event graphs there are for a collection. Networks are just to transfer event graphs to quadriga.